festim/hydrogen_transport_problem.py

```diff
@@ -123,11 +129,50 @@ def initialise(self):
         self.t = fem.Constant(self.mesh.mesh, 0.0)
         self.dt = self.settings.stepsize.get_dt(self.mesh.mesh)
 
+        self.define_temperature()
         self.define_boundary_conditions()
         self.create_formulation()
         self.create_solver()
         self.defing_export_writers()
 
+    def define_temperature(self):
+        """If temperature value is given only dependent on t, create a
+        fem.Constant else create an expression to be updated and a function"""
+        # if temperature is a float or int, create a fem.Constant
+        if isinstance(self.temperature, (float, int)):
+            self._temperature_fenics_value = F.as_fenics_constant(
+                self.temperature, self.mesh.mesh
+            )
+        elif isinstance(self.temperature, fem.Constant):
```

Comment:
I think we also want to treat the case where it's given as a `fem.Function`
```suggestion
        elif isinstance(self.temperature, (fem.Constant, fem.Function)):
```